checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakTransfer.java

```diff
@@ -0,0 +1,276 @@
+        elseStore.clearValue(targetExpr);
+        elseStore.insertValue(targetExpr, defaultType);
+      } else {
+        CFStore store = result.getRegularStore();
+        store.clearValue(targetExpr);
+        store.insertValue(targetExpr, defaultType);
+      }
+    }
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitObjectCreation(
+      ObjectCreationNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitObjectCreation(node, input);
+    updateStoreWithTempVar(result, node);
+    return result;
+  }
+
+  /**
+   * This method either creates or looks up the temp var t for node, and then updates the store to
+   * give t the same type as node
+   *
+   * @param node the node to be assigned to a temporal variable
+   * @param result the transfer result containing the store to be modified
+   */
+  public void updateStoreWithTempVar(TransferResult<CFValue, CFStore> result, Node node) {
+    // Must-call obligations on primitives are not supported.
+    if (!TypesUtils.isPrimitiveOrBoxed(node.getType())) {
+      MustCallAnnotatedTypeFactory mcAtf =
+          rlTypeFactory.getTypeFactoryOfSubchecker(MustCallChecker.class);
+      LocalVariableNode temp = mcAtf.getTempVar(node);
+      if (temp != null) {
+        rlTypeFactory.addTempVar(temp, node.getTree());
+        JavaExpression localExp = JavaExpression.fromNode(temp);
+        AnnotationMirror anm =
+            rlTypeFactory
+                .getAnnotatedType(node.getTree())
+                .getAnnotationInHierarchy(rlTypeFactory.top);
+        insertIntoStores(result, localExp, anm == null ? rlTypeFactory.top : anm);
+      }
+    }
+  }
+
+  private AnnotationMirror getUpdatedCalledMethodsType(
+      AnnotatedTypeMirror currentType, String... methodNames) {
+    AnnotationMirror type;
+    if (currentType == null || !currentType.isAnnotatedInHierarchy(rlTypeFactory.top)) {
+      type = rlTypeFactory.top;
+    } else {
+      type = currentType.getAnnotationInHierarchy(rlTypeFactory.top);
+    }
+
+    // Don't attempt to strengthen @CalledMethodsPredicate annotations, because that would
+    // require reasoning about the predicate itself. Instead, start over from top.
+    if (AnnotationUtils.areSameByName(
+        type, "org.checkerframework.checker.calledmethods.qual.CalledMethodsPredicate")) {
+      type = rlTypeFactory.top;
+    }
+
+    if (AnnotationUtils.areSame(type, rlTypeFactory.bottom)) {
+      return null;
+    }
+
+    List<String> currentMethods =
+        AnnotationUtils.getElementValueArray(
+            type, rlTypeFactory.calledMethodsValueElement, String.class);
+    List<String> newList =
+        Stream.concat(Arrays.stream(methodNames), currentMethods.stream())
+            .collect(Collectors.toList());
+
+    AnnotationMirror newType = rlTypeFactory.createCalledMethods(newList.toArray(new String[0]));
+    return newType;
+  }
+
+  private void handleEnsuresCalledMethodVarArgs(
```

Comment:
@kelloggm similarly I think this logic can be moved to the main Called Methods Checker